### ISSUE TITLE
DataTable - `onSelect` checkbox in header cell causes accessibility violation

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#6932]
The CheckBox header cell in DataTable when using `onSelect` violates the following accessibility rule:
https://dequeuniversity.com/rules/axe/4.5/empty-table-header

All header cells are required to have text in them.

### URL, screen shot, or Codepen exhibiting the issue

![](https://files.slack.com/files-pri/T04LMHMUT-F05QEKC8VNH/screen_shot_2023-08-31_at_1.10.22_pm.png)

### Steps to Reproduce

This can be tested by running the Wave accessibility tool on the [DataTable/Controlled](https://storybook.grommet.io/?path=/story/visualizations-datatable-controlled--controlled-data-table) story

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by jcfilben** [comment]
Switched from bug to enhancement since this issue isn't a wcag violation but rather an encouraged deque best practice
